### PR TITLE
Add integration test for deployment rolling update, rollback, rollover

### PR DESCRIPTION
**Pull request by janetkuo**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: ref #52113

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/integration/deployment/util.go [hunk 1]
```diff
@@ -206,40 +207,50 @@ func markPodReady(c clientset.Interface, ns string, pod *v1.Pod) error {
 	return err
 }
 
-// markUpdatedPodsReady manually marks updated Deployment pods status to ready
-func (d *deploymentTester) markUpdatedPodsReady() {
+// markUpdatedPodsReady manually marks updated Deployment pods status to ready,
+// until the deployment is complete
+func (d *deploymentTester) markUpdatedPodsReady(wg *sync.WaitGroup) {
```
Comment (kow3ns) [inline test/integration/deployment/util.go]: Why are we using a separate goroutine? It appears that the main routine just waits on wg.

File: test/integration/deployment/util.go [hunk 2]
```diff
@@ -206,40 +207,50 @@ func markPodReady(c clientset.Interface, ns string, pod *v1.Pod) error {
...
-		return readyPods >= *d.deployment.Spec.Replicas, nil
+		return false, nil
 	})
 	if err != nil {
 		d.t.Fatalf("failed to mark updated Deployment pods to ready: %v", err)
 	}
 }
 
+func (d *deploymentTester) deploymentComplete() (bool, error) {
+	latest, err := d.c.ExtensionsV1beta1().Deployments(d.deployment.Namespace).Get(d.deployment.Name, metav1.GetOptions{})
```
Comment (kow3ns) [inline test/integration/deployment/util.go]: Do we need to check ObservedGeneration to ensure ordering?

Comment (janetkuo) [reply test/integration/deployment/util.go]: `deploymentutil.DeploymentComplete()` does that

**Comment by janetkuo** [comment]
The flake is #54095

**Pull request merged**